### PR TITLE
Added button component

### PR DESCRIPTION
**Pull request by asebexen**
| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 5753e1e0d61b950c7bafa19c09054b01af4db2b8  | 
|--------|--------|

### Summary:
Added a new Button component with various styles and states, updated Storybook configuration, and removed old story files.

**Key points**:
- Added `Button` component in `src/stories/button/Button.tsx` with props for text, disabled, loading, tooltip, icon, size, and style.
- Created `Button` stories in `src/stories/button/Button.stories.ts` for primary, secondary, and tertiary styles.
- Added `button.module.sass` for button styling.
- Updated `.storybook/preview.ts` to set default background to dark and layout to centered.
- Removed old story files: `Button.stories.ts`, `Button.tsx`, `Header.stories.ts`, `Header.tsx`, `Page.stories.ts`, `Page.tsx`, `Configure.mdx`, `button.css`, `header.css`, `page.css`.
- Updated `package.json` to move `sass` from dependencies to devDependencies.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: src/stories/button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  tooltip?: string;
+  children?: string;
+  iconUrl?: string;
+  iconPosition?: 'left' | 'right',
+  size?: 'lg' | 'md' | 'sm',
+  style?: 'primary' | 'secondary' | 'tertiary';
+};
+const Button = (props: ButtonProps) => {
+  const {
```
Comment (jakezegil) [inline src/stories/button/Button.tsx:16]: nit: destructure this on line 15

File: src/stories/button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    style = 'primary',
+    onClick,
+    size = 'md',
+    tooltip,
+    iconUrl,
+    iconPosition = 'left',
+  } = props;
+
+  // Constructing button props
+  const classNameButtonArray = [styles.button, styles[style], styles[size]];
```
Comment (jakezegil) [inline src/stories/button/Button.tsx]: no push

```tsx
const buttonClassname = [
 styles.button, 
 styles[style], 
 styles[size],
 iconPosition === 'right' && styles.reverse,
 loading && styles.loading
].filter(item => !!item).join(" ")

File: src/stories/button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  const classNameButton = classNameButtonArray.join(' ');
+  const htmlButtonProps = {
+    className: classNameButton,
+    onClick,
+    disabled: disabled || loading,
+    title: tooltip
+  };
+
+  // Constructing icon element
+  const classNameIconArray = [styles.icon, styles[size]];
```
Comment (jakezegil) [inline src/stories/button/Button.tsx]: same as above

File: src/stories/button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  if (disabled || loading) classNameIconArray.push(styles.disabled);
+  const classNameIcon = classNameIconArray.join(' ');
+  const iconElement = iconUrl ? <img className={classNameIcon} src={iconUrl}/> : <></>;
```
Comment (jakezegil) [inline src/stories/button/Button.tsx:49]: nit: you can render null instead of a fragment

**Review by jakezegil — APPROVED** [review]
address feedback and approved

File: src/stories/button/Button.tsx (hunk 1 above)
Comment (asebexen) [reply src/stories/button/Button.tsx:16]: I think the case we ran into during the next PR review is compelling reason to use this as a default, unless there's a strong argument in the other direction. (Eg: Does having even one layer of prop drilling indicate a code smell?)

**Pull request merged**